### PR TITLE
Added getReviewSummary() to Mage_Catalog_Model_Product

### PR DESCRIPTION
**Pull request by luigifab**
### Description

This PR move the load of `review/review_summary` to the product model.
I don't remember exactly, but I think for us it reduce number of SQL queries in `review_entity_summary`.

OpenMage 20.0.16 / PHP 8.0.25

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Catalog/Model/Product.php [hunk 1]
```diff
@@ -2312,4 +2317,18 @@ public function getEvent()
...
+     */
+    public function getReviewSummary($storeId = null)
+    {
+        if (empty($this->_reviewSummary)) {
+            $this->_reviewSummary = Mage::getModel('review/review_summary')
+                ->setStoreId($storeId ?? Mage::app()->getStore()->getId())
+                ->load($this->getId());
+        }
+        return $this->_reviewSummary;
+    }
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Product.php:2333]: It makes more sense for me to not touch `getEntitySummary` in `app/code/core/Mage/Review/Model/Review.php`, and if we set `$_reviewSummary = []`, then we can do this:
```php
    public function getReviewSummary($storeId = 0)
    {
       if (!array_key_exists($storeId, $this->_reviewSummary)) {
            Mage::getModel('review/review')->getEntitySummary($this, $storeId);
            $this->_reviewSummary[$storeId] = $this->getRatingSummary();
       }
    }
```

I didn't test the above code.

File: app/code/core/Mage/Catalog/Model/Product.php [hunk 2]
```diff
@@ -340,6 +340,11 @@ class Mage_Catalog_Model_Product extends Mage_Catalog_Model_Abstract
      */
     protected $_calculatePrice = true;
 
+    /**
+     * @var Mage_Review_Model_Review_Summary
+     */
+    protected $_reviewSummary;
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Product.php:346]: For multi-stores:
```suggestion
     * @var Mage_Review_Model_Review_Summary[]
     */
    protected $_reviewSummary = [];
```

Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product.php:346]: :+1:

File: app/code/core/Mage/Catalog/Model/Product.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product.php:2333]: :+1:

**Comment by fballiano** [comment]
@luigifab could you check @kiatng's code?

**Comment by luigifab** [comment]
Yes I will do that.

**Comment by luigifab** [comment]
I remember why I moved the logic, this is because in our product page we have multiple templates that use `getReviewSummary()` data. Because `$this->getProduct()` or `Mage::registry('current_product')` is available everywhere, I did these changes.

You are right for multi store, I adding.

**Comment by fballiano** [comment]
This PR needs to be rebased on "main" and then we should quickly decide to accept it or not.

**Comment by fballiano** [comment]
rebased to main, dunno how it gives these many conflicts but anyway... please check the unresolved conversation so that we can move on with this.

**Comment by luigifab** [comment]
So the merge is ok.

The idea here is to allow to `$product->getReviewSummary()` from multiple template files, by loading the summary only one time, and because we can `Mage::registry('current_product')`, it works. Perhaps I'm tired, but not sure how I can do it in another way.

**Review by fballiano — APPROVED** [review]
it seems all requests were taken care of thus I'm approving

**Review by kiatng — APPROVED** [review]

**Pull request merged**